### ISSUE TITLE
Supply a custom media type with my HAL+JSON-compatible resource

### ISSUE THREAD

**Issue by mkhl** [spring-projects/spring-hateoas#1051]
I'm using Spring 5.1.8, Spring Boot 2.1.6, and Spring HATEOAS 0.25.1.

For some resource, I essentially want to do what spring boot actuator does: serve my resources with a custom media type by default, but keep the responses HAL+JSON-compatible.

For that to work, it seems like I need to essentially duplicate `ConverterRegisteringWebMvcConfigurer`, or alternatively ensure that _it_ runs before my own `WebMvcConfigurer` so I can find and reuse the custom `ObjectMapper` with the `Jackson2HalModule` already registered, then insert my custom `HttpMessageConverter` in front of the one it inserts.

```java
@Configuration
class CustomHalConfiguration implements WebMvcConfigurer {
    private final Map<Class<?>, List<MediaType>> types = Map.of(…, List.of(…));

    private final ObjectFactory<ConverterRegisteringWebMvcConfigurer> configurer;

    HalConfiguration(ObjectFactory<ConverterRegisteringWebMvcConfigurer> configurer) {
        this.configurer = configurer;
    }

    @Override
    public void extendMessageConverters(List<HttpMessageConverter<?>> converters) {
        // ensure the spring-hateoas configurer runs first so we can find its extended ObjectMapper
        configurer.getObject().extendMessageConverters(converters);
        findMapper(converters).map(this::converters)
                .ifPresent(custom -> converters.addAll(0, custom));
    }

    private Optional<ObjectMapper> findMapper(List<HttpMessageConverter<?>> converters) {
        return converters.stream()
                .filter(MappingJackson2HttpMessageConverter.class::isInstance)
                .map(MappingJackson2HttpMessageConverter.class::cast)
                .map(MappingJackson2HttpMessageConverter::getObjectMapper)
                .filter(Jackson2HalModule::isAlreadyRegisteredIn)
                .findAny();
    }

    private List<MappingJackson2HttpMessageConverter> converters(ObjectMapper mapper) {
        return types.entrySet().stream()
                .map(entry -> converter(entry.getKey(), entry.getValue()))
                .peek(converter -> converter.setObjectMapper(mapper))
                .collect(toList());
    }

    private MappingJackson2HttpMessageConverter converter(Class<?> type, List<MediaType> mediaTypes) {
        var converter = new TypeConstrainedMappingJackson2HttpMessageConverter(type);
        converter.setSupportedMediaTypes(mediaTypes);
        return converter;
    }
}
```

(The snippet probably overuses the Stream API, please let me know if that distracts from the issue.)

This configuration feels more like I'm fighting the API than using it, is there an easier way ot accomplish what I want that I'm missing?

If there isn't, it would be _really_ nice if there were a better way to extend the HAL+JSON-compatible `HttpMessageConverter` or provide custom ones, or maybe just if there were a way to access the HAL-enabled `ObjectMapper` that `ConverterRegisteringWebMvcConfigurer` constructs.

**Comment by gregturn** [comment]
This is something we implemented in 1.0. Backporting to 0.25 would require sweeping changes.

**Comment by mkhl** [comment]
I see, thank you for letting me know.
